__tests__/app.test.js

```
@@ -26,129 +22,159 @@ describe("GET /api", () => {
-        expect(typeof article.comment_count).toEqual("number")
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body
+        expect(articles.length).toBe(13)
+        expect(articles).toBeSortedBy("created_at", { descending: true })
+        articles.forEach((article) => {
+          expect(typeof article.article_id).toEqual("number");
+          expect(typeof article.title).toEqual("string");
+          expect(typeof article.topic).toEqual("string");
+          expect(typeof article.author).toEqual("string");
+          expect(typeof article.created_at).toEqual("string");
+          expect(typeof article.votes).toEqual("number");
+          expect(typeof article.article_img_url).toEqual("string");
+          expect(typeof article.comment_count).toEqual("number")
+        })
       })
+  })
+})
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("Responds with all comments under the given article", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body
+        expect(comments.length).toBe(2)
+        expect(comments).toBeSortedBy("created_at", { descending: false })
+        comments.forEach((comment) => {
+          expect(typeof comment.comment_id).toBe("number")
+          expect(typeof comment.body).toBe("string"),
+            expect(typeof comment.votes).toBe("number"),
+            expect(typeof comment.author).toBe("string"),
+            expect(typeof comment.created_at).toBe("string"),
+            expect(comment.article_id).toBe(3)
+        })
+      })
+  })
+  test("Responds with a 404 if either the article doesn't exist or there are no comments associated", () =>{
+    return request(app)
+    .get("/api/articles/999/comments")
+    .expect(404)
+    .then(({ body }) => {
+      const { msg } = body
+      expect(msg).toBe("No comments found")
```

If the article_id is valid, but there are no comments yet (e.g. the article is newly posted), we won't want to throw an error. From the client's perspective, an empty array will be useful information.You can use one of your existing models to implemenet this behaviour 👀 
 I'd add a separate test here: 
`200: Responds with an empty array if article_id is valid but there are no associated comments`